cmd/simulator/load/loader.go

```diff
@@ -29,6 +43,20 @@ func ExecuteLoader(ctx context.Context, config config.Config) error {
 		defer cancel()
 	}
 
+	// Create buffered sigChan to send SIGINT notifications
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT)
+
+	// Create context with cancel
+	ctx, cancel := context.WithCancel(ctx)
+
+	go func() {
+		// Blocks until we receive a SIGNINT notification
+		<-sigChan
+		// Cancel the context and end all shared processes
+		cancel()
+	}()
```

Comment:
Seems that this could leak if context was done/cancelled before sigInt. So we should probably return if ctx was done. See: https://gist.github.com/matejb/87064825093c42c1e76e7175665d9a9b